### PR TITLE
Resize T5 Vocab

### PR DESCRIPTION
**Pull request by dirkgr**
CC @HarshTrivedi

### CODE REVIEW

**Comment by HarshTrivedi** [comment]
Thank you! @dirkgr

File: allennlp/modules/transformer/t5.py [hunk 1]
```diff
@@ -806,6 +835,35 @@ def __init__(
...
+    def resize_token_embeddings(
+        self, new_size: int, *, init_fn: Callable = torch.nn.init.normal_
+    ) -> None:
+        self.encoder.resize_token_embeddings(new_size, init_fn=init_fn)
+        # If encoder and decoder share embeddings, this is a no-op the second time.
+        self.decoder.resize_token_embeddings(new_size, init_fn=init_fn)
+
+        # resize lm head
+        old_size = self.lm_head.out_features
+        if old_size == new_size:
```
Comment (AkshitaB) [inline allennlp/modules/transformer/t5.py:847]: Minor thing: Maybe we can do this check first thing? It'll avoid the 2 empty calls.

Comment (AkshitaB) [reply allennlp/modules/transformer/t5.py:847]: Or perhaps not? We want to resize the embedding dim and the output dim to the same `new_size`. Will the old `embedding_dim` and the old `lm_head.out_features` be different?

**Review by AkshitaB — CHANGES_REQUESTED** [review]
Left a couple comments. Looks fine to me, overall. Will `resize_token_embeddings` be called internally somewhere? When does the user call it? Should there be a docstring for what `init_fn` is supposed to be (it wasn't clear to me without reading the code)?

File: allennlp/modules/transformer/t5.py (hunk 1 above)
Comment (dirkgr) [reply allennlp/modules/transformer/t5.py:859]: It's a bit of defensive programming. It should never happen, but if the three places where the size of the token embedding matters get out of sync, this call will sync them all back up.

**Review by dirkgr — COMMENTED** [review]
I added a docstring in 074975f181ac1aa1c9c9d202af99a41f7c5728bd, to hopefully make it clearer what `init_fn` does.

**Comment by dirkgr** [comment]
I also want to point out that I managed to push this and it passed all CI checks the first time 😁.

**Review by AkshitaB — APPROVED** [review]
